### DISCUSSION TITLE
How to suppress "command not found" if I use on_command_not_found event?

### DISCUSSION THREAD

**Discussion by kellerkind84** [xonsh/xonsh#5277] [Q&A] [answered]
Hi everybody!

So, I have an idea... basically, I want a third mode besides Python and subprocess, where if I just enter some plain text, the input is simply fed to a llm. It seems rather simple to me, if it's not python, and the subprocess can't find the command, then my own function is executed. 
So far I've tried to simply use the on_command_not_found event and simply hand the input through, which works quite well. The obvious drawback is that it still prints the "xonsh: subprocess mode: command not found" error. I don't really know how to suppress that in an event handler.
Any ideas what I could do here?

Thanks already!

**Comment by anki-code** [comment]
Here is the code:

https://github.com/xonsh/xonsh/blob/1d54c74832ad76073e481fe668b89bdb450989bb/xonsh/procs/specs.py#L490-L493

Probably you can improve this logic e.g. control the error using the on_command_not_found event.

Feel free to wrap your idea into xontrib using [xontrib-template](https://github.com/xonsh/xontrib-template) in 5 minutes.

**Reply by kellerkind84** [reply]
Thank you for the quick reply!

Yeah that code is exactly what I was looking for, but it also illustrates my problem. The error that is raised there is not associated with the event, so in my understanding I cannot directly handle this error myself in my code, right?

I also don't know of any way to change the error handling in xonsh via configuration... except for traceback.

So basically, as I see it right now I'm left with changing the xonsh source itself, right?

**Comment by anki-code** [comment, marked as answer]
Yes. You should propose changes and the PR to implement them.
For example this logic could be: if on_command_not_found event returns 0, skip showing the error.
